File: JPASandBar/src/main/java/com/downstreammedia/sandbar/entities/Location.java
package com.downstreammedia.sandbar.entities;

import java.util.Objects;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;


@Getter
@Setter
@Accessors(chain=true)
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Location {
	
	/*********FIELDS*********/
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	
	private String name;
	
	private String description;
	
	private String address;
	
	private String city;
	
	private int zip;
	
	private String latitude;

	private String longitude;

	/*********METHODS*********/
	@Override
	public int hashCode() {
		return Objects.hash(id, latitude, longitude, name);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Location other = (Location) obj;
		return id == other.id && Objects.equals(latitude, other.latitude) && Objects.equals(longitude, other.longitude)
				&& Objects.equals(name, other.name);
	}

	@Override
	public String toString() {
		StringBuilder builder = new StringBuilder();
		builder.append("Location [id=");
		builder.append(id);
		builder.append(", name=");
		builder.append(name);
		builder.append(", description=");
		builder.append(description);
		builder.append(", address=");
		builder.append(address);
		builder.append(", city=");
		builder.append(city);
		builder.append(", zip=");
		builder.append(zip);
		builder.append(", latitude=");
		builder.append(latitude);
		builder.append(", longitude=");
		builder.append(longitude);
		builder.append("]");
		return builder.toString();
	}
}
	

File: ngTripLeader/ngTripLeader/src/app/models/location.ts

export class Location {

  id: number;
  name: string;
  description: string;
  address: string;
  city: string;
  zip: number;
  latitude: string;
  longitude: string;

  constructor(
    id?: number,
    name?: string,
    description?: string,
    address?: string,
    city?: string,
    zip?: number,
    latitude?: string,
    longitude?: string

  ){

    if (!id) {
      throw new Error("Unexpected error: Location missing id");
  }
    this.id = id;

    if (!name) {
      throw new Error("Unexpected error: Location missing name");
  }
    this.name = name;

    if (!description) {
      throw new Error("Unexpected error: Location missing description");
  }
    this.description = description;

    if (!address) {
      throw new Error("Unexpected error: Location missing address");
  }
    this.address = address;

    if (!city) {
      throw new Error("Unexpected error: Location missing city");
  }
    this.city = city;

    if (!zip) {
      throw new Error("Unexpected error: Location missing zip");
  }
    this.zip = zip;

    if (!latitude) {
      throw new Error("Unexpected error: Location missing latitude");
  }
    this.latitude = latitude;

    if (!longitude) {
      throw new Error("Unexpected error: Location missing longitude");
  }
    this.longitude= longitude;
  }



}
